snuba/utils/schemas.py

```diff
@@ -702,3 +705,41 @@ def set_modifiers(self, modifiers: Optional[TModifiers]) -> Enum[TModifiers]:
 
     def get_raw(self) -> Enum[TModifiers]:
         return Enum(self.values)
+
+
+class ColumnValidator:
+    def __init__(self, column_set: ColumnSet):
+        self._column_set = column_set
+
+    def validate(self, column_name: str, values: Sequence[AnyType]) -> None:
+        expected_type = self._column_set[column_name].type
+        val_func: Optional[Callable[[AnyType], None]]
+        match expected_type:
+            case UUID():
+                val_func = self._validate_uuid
+            case Int():
+                val_func = self._validate_int
+            case UInt():
+                val_func = self._validate_int
+            case Float():
+                val_func = self._validate_float
+            case String():
+                val_func = self._validate_string
+            case _:
+                raise NotImplementedError(
+                    f"ColumnValidator not implement for: {expected_type}"
+                )
+        for val in values:
+            val_func(val)
+
+    def _validate_uuid(self, value: str) -> None:
+        assert uuid.UUID(str(value))
```

Comment:
is it fine for these to be asserts or should they be exceptions raised?

Reply:
I actually might change these all to return a bool and then if it's false in the main validate function i'll raise a invalid column type exception